src/nprintml/cli.py

```diff
@@ -1,20 +1,147 @@
-        # TODO: but, if that's not the case let's get rid of this:
-        # formatter_class=argparse_formatter.ParagraphFormatter,
+        print(f'error:{exc.__class__.__module__}.{exc.__class__.__name__} ✕')
+        sys.exit(1)
 
+
+def build_parser(**parser_kwargs):
+    """Construct base parser & arguments for nprintml.
+
+    Link back to parser is set on any resulting parsed argument
+    `Namespace` as `__parser__`.
+
+    """
+    parser = argparse.ArgumentParser(
+        description='train models for network traffic analysis',
         **parser_kwargs,
     )
-    args = parser.parse_args(argv)
 
-    raise NotImplementedError(args)
+    version = f'nprintML {nprintml.__version__} | nPrint {nprintml.__nprint_version__}'
+    # support published command aliases
+    prog = pathlib.Path(sys.argv[0])
+    if prog.name != 'nprintml' and prog.suffix != '.py':
+        version = f'%(prog)s {nprintml.__version__} | {version}'
+
+    parser.add_argument(
+        '--version',
+        action='version',
+        help="show program version and exit",
+        version=version,
+    )
+
+    parser.add_argument(
+        '-V', '--verbose',
+        action='store_true',
+        help="print human readable packets with nPrints",
+    )
+
+    parser.add_argument(
+        '--tb', '--traceback',
+        action='store_true',
+        dest='traceback',
+        help="print exception tracebacks",
+    )
+
+    output_default = get_default_directory()
+    parser.add_argument(
+        '-o', '--output',
+        default=output_default,
+        dest='outdir',
+        metavar='DIR',
+        type=pathlib.Path,
+        help="output directory path to which to write artifacts and results "
+             f"(default: {output_default})",
+    )
+
+    parser.set_defaults(
+        __parser__=parser,
+    )
+
+    return parser
+
+
+def pairwise(iterable):
+    """s -> (s0, s1), (s1, s2), (s2, s3), ..., (sn, None)"""
+    (a, b) = itertools.tee(iterable)
+    next(b, None)
+    return itertools.zip_longest(a, b)
+
+
+def get_default_directory(base_name='nprintml', words=ANIMALS):
+    """Construct user-friendly default output directory path."""
+    path = pathlib.Path(base_name)
+
+    if path.exists():
+        word_options = '|'.join(words)
+        run_pattern = re.compile(rf'run-({word_options})-', re.I)
+        run_matches = (run_pattern.match(run_path.name) for run_path in path.glob('run-*'))
+        words_used = sorted(
+            (run_match.group(1).lower() for run_match in run_matches if run_match),
+            reverse=True,
+        )
+
+        if words_used:
+            last_word = words_used[0]
+            for (word0, word1) in pairwise(words):
+                if word0 == last_word:
+                    next_word = word1 or words[0]
+                    break
+        else:
+            next_word = words[0]
+    else:
+        next_word = words[0]
+
+    return path / f'run-{next_word}-{int(time.time())}-{os.getpid()}'
```

Comment:
I gave this some thought, though I wouldn't be surprised if we end up wanting to tweak it.

With the output directory setup we're currently imagining, it seemed cleanest to ensure that -- by default at least -- we put things into an obvious subdirectory -- `nprintml/` -- _and_ relieve users of the challenges of handling iterative runs.

The numeric part of the naming handles date metadata and sort order, and thread-safety; but, it also occurred to me how annoying it can be to tab-complete directories named that way. So the alphabetical animal names are an attempt at making that friendlier.